src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/AbstractHostObject.cs

```diff
@@ -7,12 +7,22 @@
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
 {
+
     /// <summary>
     /// Base type for language service host object for cross targeting projects.
     /// </summary>
-    internal abstract class AbstractHostObject : IVsHierarchy, IVsContainedLanguageProjectNameProvider
+    internal abstract class AbstractHostObject : IVsHierarchy, IVsContainedLanguageProjectNameProvider, IVsProject
     {
-        protected abstract IVsHierarchy InnerHierarchy { get; }
+        protected AbstractHostObject(object innerHostObject)
```

Comment:
Should we have the constructor here take an IVsHierarchy and IVsProject, instead of just an object? That way, if we ever have to implement this type again, we know explicitly what is required, and the implementing type can do the cast itself. Otherwise, I'd like to see what this object should be documented somewhere, so I don't have to look into the implementation of this object to know what needs to be passed.